xain_sdk/participant_state_machine.py

```
@@ -378,12 +379,15 @@ def start_participant(participant: Participant, config: Config) -> None:
 
         # in WAITING state
         logger.info("rendezvous successful, begin WAITING...")
-        begin_waiting(state_record, channel, internal_participant)
-
-        # in DONE state
-        logger.info("shutting down participant...")
-        terminate.set()
-        msg_loop.join()
+        try:
+            begin_waiting(state_record, channel, internal_participant)
+        except Exception as err:
```

Just want to mention that as an alternative we could just let it crash:

```python
try:
    begin_waiting()
finally:
    terminate.set()
    # ...
```

I don't see any advantage/drawback to either solution though the current exception handling looks good to me as well.

---

Yeah maybe it is better to let it crash because then it is obvious to the caller that something went wrong.